pkg/util/alibaba/alibaba.go

```diff
@@ -33,6 +33,9 @@ func IsRunningOn() bool {
 
 // GetHostAlias returns the VM ID from the Alibaba Metadata api
 func GetHostAlias() (string, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
+		return "", nil
```

Comment:
I missed this: we should return an error here, the caller should only have to check the error. Same for all other call  to  `IsCloudProviderEnabled`.

Sorry I missed it.

Reply:
Looking over this again, I think your hunch may have been right the first time.  The error messages I have here are at the DEBUG log level currently, and are set to print at debug level if the cloud_provider is or is not set.  I'd like to leave this for now, as I think these message probably belong in debug logs.

Reply:
My point is that the code callinh GetHostAlias or GetHostname will check the error not the value. We need to return an error here and not nil.

For example https://github.com/DataDog/datadog-agent/blob/master/pkg/util/hostname.go#L167 will call this https://github.com/DataDog/datadog-agent/blob/master/pkg/util/hostname.go#L167 whi would return an empty string and nil. Since we only check the error (https://github.com/DataDog/datadog-agent/blob/master/pkg/util/hostname.go#L169) we would set the hostname as an empty string.

Unless I missing something and error should be raise in all function that return nil right now when disabled.